lib/services/api/session_manager.ts

```diff
@@ -283,8 +286,16 @@ export class SNSessionManager extends PureService {
     const user = response.user;
     this.user = user;
     await this.storageService!.setValue(StorageKey.User, user);
-    const session = new Session(response.token);
+    const session = this.newSessionFromResponse(response);
     await this.storageService!.setValue(StorageKey.Session, session);
     await this.setSession(session);
   }
+
+  private newSessionFromResponse(response: HttpResponse) {
```

Comment:
This is a duplication of code from api_service. This function looks like it should be a static method in the Session class.